contracts/HATVault.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+// Disclaimer https://github.com/hats-finance/hats-contracts/blob/main/DISCLAIMER.md
+
+pragma solidity 0.8.14;
+
+
+import "./vaults/Claim.sol";
+import "./vaults/Deposits.sol";
+import "./vaults/Params.sol";
+import "./vaults/Withdrawals.sol";
+
+contract HATVault is Claim, Deposits, Params, Withdrawals {
+
+    function _deposit(
+        address caller,
+        address receiver,
+        uint256 assets,
+        uint256 shares
+    ) internal override(Deposits, ERC4626Upgradeable) {
+        Deposits._deposit(caller, receiver, assets, shares);
```

Comment:
why copy the function here?

Reply:
It won't compile otherwise. That's another reason I wanted to scrap the multiple files thing and just have it all in one file (also I think the code now is not too long for that anymore).